File: App Dev Based Projects/CurrencyConverter/app/src/main/java/com/example/currencyconverter/MainActivity.java
package com.example.currencyconverter;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    public void convert(View view)
    {
        EditText amountInInr = (EditText) findViewById(R.id.amountInInr);
        double inUSD = Double.parseDouble(amountInInr.getText().toString());
        inUSD = inUSD*72.51;
        String ans = String.format("%.2f",inUSD);
        Toast.makeText(this, amountInInr.getText().toString()+" in USD is "+ans, Toast.LENGTH_SHORT).show();
    }
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
}

File: App Dev Based Projects/AgeInMinutes/app/src/main/java/com/atulyadav/ageinminutes/MainActivity.kt
package com.atulyadav.ageinminutes

import android.app.DatePickerDialog
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import java.text.SimpleDateFormat
import java.util.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btndatepicker.setOnClickListener {
            clickDatePicker()
        }
    }

    private fun clickDatePicker() {
        val myCalendar = Calendar.getInstance()
        val year = myCalendar.get(Calendar.YEAR)
        val month = myCalendar.get(Calendar.MONTH)
        val day = myCalendar.get(Calendar.DAY_OF_MONTH)
        val dpd = DatePickerDialog(
            this, { _, Selectedyear, Selectedmonth, SelecteddayOfMonth ->
                val selectedDate = "$SelecteddayOfMonth/ ${Selectedmonth + 1}/ $Selectedyear"

                tvSelectedDate.text = selectedDate

                val sdf = SimpleDateFormat("dd/MM/yyyy", Locale.ENGLISH)

                val theDate = sdf.parse(selectedDate)

                val selectedDateInMinutes = theDate!!.time / 60000

                val currentDate = sdf.parse(sdf.format(System.currentTimeMillis()))
                val currentDateInMinutes = currentDate!!.time / 60000

                val differenceInMinutes = currentDateInMinutes - selectedDateInMinutes

                tvSelectedDateInMinute.text = differenceInMinutes.toString()


            }, year, month,
            day
        )
        dpd.datePicker.maxDate = Date().time - 86400000
        dpd.show()
    }

}

